### ISSUE TITLE
Receiving readHandshakeRecord when running diag

### ISSUE THREAD

**Issue by joshFive** [elastic/support-diagnostics#466]
Trying to run diag against es 6.8 running in a docker container.

using openJDK 1.8

command: ./diagnostics.sh --host lpdosput020008.phx.aexp.com -u elastic -p --port 9200 --ssl --type api --noVerify --bypassDiagVerify

When running the CLI we receive these errors:
```
Could not retrieve the Elasticsearch version due to a system or network error - unable to continue. readHandshakeRecord
```
```
main ERROR Unable to locate appender "diag" for logger config "root"
```

Here is the error log from the bundle:

```
11:39:01.516 [main] INFO  com.elastic.support.BaseService - Diagnostic logger reconfigured for inclusion into archive
11:39:01.518 [main] INFO  com.elastic.support.diagnostics.commands.CheckElasticsearchVersion - Getting Elasticsearch Version.
11:39:01.780 [main] ERROR com.elastic.support.rest.RestClient - Unexpected Execution Error
javax.net.ssl.SSLException: readHandshakeRecord
	at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1063) ~[?:1.8.0_275]
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:394) ~[?:1.8.0_275]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.10.jar:4.5.10]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72) ~[httpclient-4.5.10.jar:4.5.10]
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:73) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execGet(RestClient.java:68) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execQuery(RestClient.java:58) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.commands.CheckElasticsearchVersion.getElasticsearchVersion(CheckElasticsearchVersion.java:76) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.commands.CheckElasticsearchVersion.execute(CheckElasticsearchVersion.java:60) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.chain.DiagnosticChainExec.runDiagnostic(DiagnosticChainExec.java:22) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.DiagnosticService.exec(DiagnosticService.java:68) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.DiagnosticApp.main(DiagnosticApp.java:42) [support-diagnostics-8.1.2.jar:8.1.2]
	Suppressed: java.net.SocketException: Broken pipe (Write failed)
		at java.net.SocketOutputStream.socketWrite0(Native Method) ~[?:1.8.0_275]
		at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111) ~[?:1.8.0_275]
		at java.net.SocketOutputStream.write(SocketOutputStream.java:155) ~[?:1.8.0_275]
		at sun.security.ssl.SSLSocketOutputRecord.encodeAlert(SSLSocketOutputRecord.java:81) ~[?:1.8.0_275]
		at sun.security.ssl.TransportContext.fatal(TransportContext.java:355) ~[?:1.8.0_275]
		at sun.security.ssl.TransportContext.fatal(TransportContext.java:267) ~[?:1.8.0_275]
		at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:397) ~[?:1.8.0_275]
		at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.10.jar:4.5.10]
		at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72) ~[httpclient-4.5.10.jar:4.5.10]
		at com.elastic.support.rest.RestClient.execRequest(RestClient.java:73) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.rest.RestClient.execGet(RestClient.java:68) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.rest.RestClient.execQuery(RestClient.java:58) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.diagnostics.commands.CheckElasticsearchVersion.getElasticsearchVersion(CheckElasticsearchVersion.java:76) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.diagnostics.commands.CheckElasticsearchVersion.execute(CheckElasticsearchVersion.java:60) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.diagnostics.chain.DiagnosticChainExec.runDiagnostic(DiagnosticChainExec.java:22) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.diagnostics.DiagnosticService.exec(DiagnosticService.java:68) [support-diagnostics-8.1.2.jar:8.1.2]
		at com.elastic.support.diagnostics.DiagnosticApp.main(DiagnosticApp.java:42) [support-diagnostics-8.1.2.jar:8.1.2]
Caused by: java.net.SocketException: Broken pipe (Write failed)
	at java.net.SocketOutputStream.socketWrite0(Native Method) ~[?:1.8.0_275]
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111) ~[?:1.8.0_275]
	at java.net.SocketOutputStream.write(SocketOutputStream.java:155) ~[?:1.8.0_275]
	at sun.security.ssl.SSLSocketOutputRecord.encodeChangeCipherSpec(SSLSocketOutputRecord.java:221) ~[?:1.8.0_275]
	at sun.security.ssl.OutputRecord.changeWriteCiphers(OutputRecord.java:162) ~[?:1.8.0_275]
	at sun.security.ssl.ChangeCipherSpec$T10ChangeCipherSpecProducer.produce(ChangeCipherSpec.java:118) ~[?:1.8.0_275]
	at sun.security.ssl.Finished$T12FinishedProducer.onProduceFinished(Finished.java:390) ~[?:1.8.0_275]
	at sun.security.ssl.Finished$T12FinishedProducer.produce(Finished.java:374) ~[?:1.8.0_275]
	at sun.security.ssl.SSLHandshake.produce(SSLHandshake.java:421) ~[?:1.8.0_275]
	at sun.security.ssl.ServerHelloDone$ServerHelloDoneConsumer.consume(ServerHelloDone.java:182) ~[?:1.8.0_275]
	at sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:377) ~[?:1.8.0_275]
	at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444) ~[?:1.8.0_275]
	at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:422) ~[?:1.8.0_275]
	at sun.security.ssl.TransportContext.dispatch(TransportContext.java:182) ~[?:1.8.0_275]
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:149) ~[?:1.8.0_275]
	at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1143) ~[?:1.8.0_275]
	at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1054) ~[?:1.8.0_275]
	... 20 more
11:39:01.789 [main] ERROR com.elastic.support.diagnostics.commands.CheckElasticsearchVersion - Unanticipated error:
java.lang.RuntimeException: readHandshakeRecord
	at com.elastic.support.rest.RestClient.execRequest(RestClient.java:79) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execGet(RestClient.java:68) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.rest.RestClient.execQuery(RestClient.java:58) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.commands.CheckElasticsearchVersion.getElasticsearchVersion(CheckElasticsearchVersion.java:76) ~[support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.commands.CheckElasticsearchVersion.execute(CheckElasticsearchVersion.java:60) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.chain.DiagnosticChainExec.runDiagnostic(DiagnosticChainExec.java:22) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.DiagnosticService.exec(DiagnosticService.java:68) [support-diagnostics-8.1.2.jar:8.1.2]
	at com.elastic.support.diagnostics.DiagnosticApp.main(DiagnosticApp.java:42) [support-diagnostics-8.1.2.jar:8.1.2]
11:39:01.792 [main] ERROR com.elastic.support.diagnostics.DiagnosticService - Could not retrieve the Elasticsearch version due to a system or network error - unable to continue. readHandshakeRecord
Check diagnostics.log in the archive file for more detail.
11:39:01.792 [main] INFO  com.elastic.support.BaseService - Closing loggers.
11:39:01.793 [main] INFO  com.elastic.support.BaseService - Archiving diagnostic results.
```

**Comment by pickypg** [comment]
In Java (and most languages), `Caused By` exceptions are the more important ones:

```
Caused by: java.net.SocketException: Broken pipe (Write failed)
```

The socket is failing _during_ the Read Handshake. This means there's something wrong at the network level (or the other side crashed).
